### DISCUSSION TITLE
How to correctly use --skip-domain option?

### DISCUSSION THREAD

**Discussion by pawelmalak** [Serene-Arc/bulk-downloader-for-reddit#369] [Q&A] [answered]
Hi. I'm trying to skip downloading videos from youtube but I can't get it to work correctly. I have tried:
- `--skip-domain youtube.com`,
- `--skip-domain www.youtube.com`,
- `--skip-domain https://www.youtube.com/`,
- `--skip-domain youtube`,

But neither of them seems to work. Youtube URL is formated like this: `https://www.youtube.com/watch?v=<VIDEO_ID>`

**Comment by Serene-Arc** [comment, marked as answer]
The command `python3 -m bdfr download test/ -l ncjq03 --skip-domain youtube.com` works for me. Note that some Youtube domains are shortened to youtu.be in some cases. If you think you have a submission that should be skipped but isn't, please open a bug report and we'll investigate.

**Reply by pawelmalak** [reply]
I was using this command `python -m bdfr download posts/ -l bqla85 --skip-domain youtube.com --verbose` and while it's true that file doesn't exist in posts/ directory it's still being downloaded and just removed by the filter afterwards. **I was thinking that filter would check URL first and just skip it. Is that intentional?**

Here is log from `--verbose` option:

> [2021-05-20 16:46:24,498 - bdfr.downloader - DEBUG] - Loading configuration from C:\Users\Pawel\AppData\Local\BDFR\bdfr\default_config.cfg
[2021-05-20 16:46:24,501 - bdfr.downloader - DEBUG] - Setting maximum download wait time to 120 seconds
[2021-05-20 16:46:24,501 - bdfr.downloader - DEBUG] - Setting datetime format string to ISO
[2021-05-20 16:46:24,502 - bdfr.downloader - DEBUG] - Using unauthenticated Reddit instance
[2021-05-20 16:46:25,199 - bdfr.downloader - DEBUG] - Attempting to download submission bqla85
[2021-05-20 16:46:25,200 - bdfr.downloader - DEBUG] - Using Youtube with url https://www.youtube.com/watch?v=Jxeen7_Ty8A
[2021-05-20 16:46:43,394 - bdfr.downloader - DEBUG] - Download filter removed bqla85 with URL https://www.youtube.com/watch?v=Jxeen7_Ty8A
[2021-05-20 16:46:43,417 - root - INFO] - Program complete

As you can see there is 18 second pause between `Using Youtube with url` and `Download filter removed bqla85 with URL`. I have also tested that with Windows Task Manager - Network Tab opened and after I ran the above command, the download speed immediately spiked which indicates file being downloaded.

**Reply by Serene-Arc** [reply]
Ah, yes you've run into an unfortunate consequence of how the Youtube module works. There's no way to actually only retrieve the link like with the other modules, since Youtube does not have a single link. Therefore, the download happens when the link is parsed, earlier than the rest of the resources and the data is held in memory. There is an enhancement on the way that allows you to disable entire modules which will solve this issue for sure, but I might be able to code a fix for it now. Please open a bug issue and I'll get to it soon.
